src/modules/45-dashboards/pages/home/DashboardForm.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import {
+  Button,
+  Formik,
+  FormikForm as Form,
+  FormInput,
+  Heading,
+  Layout,
+  ModalErrorHandler,
+  ModalErrorHandlerBinding,
+  PageSpinner,
+  SelectOption
+} from '@wings-software/uicore'
+import { FontVariation } from '@harness/design-system'
+import * as Yup from 'yup'
+import { useParams } from 'react-router-dom'
+import { useStrings } from 'framework/strings'
+import { useGetFolder, ErrorResponse } from 'services/custom-dashboards'
+import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
+import css from './HomePage.module.scss'
+
+const TAGS_SEPARATOR = ','
+
+interface DashboardFormProps {
+  formData?: DashboardFormRequestProps
+  title: string
+  loading: boolean
+  onComplete: (data: DashboardFormRequestProps) => void
+  modalErrorHandler?: ModalErrorHandlerBinding
+  setModalErrorHandler: (modalErrorHandler: ModalErrorHandlerBinding) => void
+}
+
+export interface DashboardFormRequestProps {
+  description?: string
+  folderId: string
+  name: string
+}
+
+export interface DashboardFormikValues {
+  description: string[]
+  folderId: string
+  name: string
+}
+
+const DashboardForm: React.FC<DashboardFormProps> = ({
+  formData,
+  title,
+  loading,
+  onComplete,
+  modalErrorHandler,
+  setModalErrorHandler
+}) => {
+  const { getString } = useStrings()
+  const { accountId } = useParams<AccountPathProps>()
+  const [initialValues, setInitialValues] = React.useState<DashboardFormikValues>({
+    folderId: formData?.folderId || '',
+    name: formData?.name || '',
+    description: formData?.description?.split(TAGS_SEPARATOR) || []
+  })
+  const sharedFolder = {
+    value: getString('dashboards.shared'),
+    label: getString('dashboards.sharedFolderTitle')
+  }
+  const [folderListItems, setFolderListItems] = React.useState<SelectOption[]>([sharedFolder])
+
+  const {
+    data: foldersList,
+    error,
+    loading: isFoldersLoading
+  } = useGetFolder({
+    queryParams: { accountId: accountId }
```

Comment:
Totally fine, but you can streamline this a wee bit
```suggestion
    queryParams: { accountId }
```